common/beerocks/bwl/dwpal/ap_wlan_hal_dwpal.cpp

```diff
@@ -2521,7 +2524,10 @@ bool ap_wlan_hal_dwpal::process_dwpal_nl_event(struct nl_msg *msg)
 std::shared_ptr<ap_wlan_hal> ap_wlan_hal_create(std::string iface_name, hal_conf_t hal_conf,
                                                 base_wlan_hal::hal_event_cb_t callback)
 {
-    return std::make_shared<dwpal::ap_wlan_hal_dwpal>(iface_name, callback, hal_conf);
+    auto nl80211_client = nl80211_client_factory::create_instance();
+    LOG_IF(!nl80211_client, FATAL) << "Failed to create nl80211_client instance";
+    return std::make_shared<dwpal::ap_wlan_hal_dwpal>(iface_name, callback, hal_conf,
+                                                      std::move(nl80211_client));
 }
```

Comment:
I moved the `create_instance` to `base_wlan_hal_XXX` constructor since the nl80211_client can be and will be used by the base class to update radio info which is implemented in the base class.

As @vitalybu said:
>Dependency injection is useful when you want to "inject" different implementations of the same interface to some class. I'm just not sure if it's the case here. Why would we ever use a non-DWPAL nl80211 client with DWPAL BWL?